### PR TITLE
250 implement react query for input title updates

### LINKED ISSUE #250: Implement React Query for Input Title Updates
**Related User Story**
#120 

**Parent Task**
#249 

**Acceptance Criteria**
- API function for updating input title is created
- Data mutation for updating input title is implemented
- Error handling with SnackBarNotification is in place
- Relevant components use the new data management method
- Appropriate caching strategy is implemented

### PR DESCRIPTION
**Pull request by kinskiwu**
## Description
- Created API function for updating input title
- Implemented data mutation for updating input title

Fixes #250 

## Type of change
- [X] ✨  New feature (non-breaking change which adds functionality)

## How Can this be tested? Testing Plan to review this PR
- [X] Where in the app does this change take place? 
UserInputList

- [X] What are the steps to test this change?
Run npm run dev
Click 'Rename' and insert a new name 'New Test'
image
- [X] What is the expected output?
image

## Checklist:

- [X] I have performed a self-review of my own code.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [X] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [X] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [X] I have ensured that my pull request title is descriptive.

## Additional Information:
Integration with UI and snack bar notification will be handled in #125.

### CODE REVIEW

**Review by colinvandergraaf — APPROVED** [review]
Looks good! But I'm curious why the whole list flickers when the update is triggered.

File: client/hooks/useUpdateInputTitleMutation.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    unknown
+  >({
+    mutationFn: updateInputTitleInApi,
+
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({ queryKey: ['inputHistory'] });
+    },
+
+    onError: (error: Error) => {
+      console.error('Error updating input title:', error);
```
Comment (relandboyle) [inline client/hooks/useUpdateInputTitleMutation.ts:52]: @kinskiwu This looks like a perfect place to trigger a snack bar alert rather than a console.log.

**Review by relandboyle — APPROVED** [review]

File: client/hooks/useUpdateInputTitleMutation.ts (hunk 1 above)
Comment (kinskiwu) [reply client/hooks/useUpdateInputTitleMutation.ts:52]: Yes it is a placeholder for now. I have another ticket to implement the snackbar #247 in bulk lol

**Comment by kinskiwu** [comment]
@colinvandergraaf I think because the onSucess triggers a re-fetch of the inputHistory so the list gets re-rendered which causes the flickering. I will take a closer look to see how I can reduce the flickering.

**Pull request merged**